src/Language/AST/NodeList.php

```diff
@@ -133,15 +134,14 @@ public function splice(int $offset, int $length, $replacement = null): NodeList
     }
 
     /**
-     * @param NodeList|array<Node|array<string, mixed>> $list
-     * @phpstan-param NodeList<T>|array<T> $list
+     * @phpstan-param iterable<T> $list
      *
      * @phpstan-return NodeList<T>
      */
-    public function merge($list): NodeList
+    public function merge(iterable $list): NodeList
     {
-        if ($list instanceof self) {
-            $list = $list->nodes;
+        if (! is_array($list)) {
+            $list = iterator_to_array($list);
```

Comment:
this will fail when key is not array-key, can we type it as `iterable<array-key, T>`?

Reply:
Good catch, never thought about this.